data-sources/discourse/src/generate-csv.ts

```diff
@@ -0,0 +1,68 @@
+const csvWriter = require('csv-writer')
+const Web3 = require('web3')
+const ethers = require('ethers')
+const Discourse = require('../abis/Discourse.json')
+
+const web3 = new Web3('https://rpc.ankr.com/eth')
+console.info('web3.version:', web3.version)
+
+const ethersProvider = new ethers.providers.JsonRpcProvider('https://rpc.ankr.com/eth')
+console.info('ethersProvider:', ethersProvider)
+
+const DiscourseContract = new web3.eth.Contract(Discourse.abi, '0xC396F3536Cc67913bbE1e5E454c10BBA4ae8928F')
+console.info('DiscourseContract._address:', DiscourseContract._address)
+
+generateData()
+
+/**
+ * Fetches data from citizens.csv and looks up the Discourse usernames.
+ */
+async function generateData() {
+  console.info('generateData')
+
+  const outputFilePath = 'output/discourse-usernames.csv'
+  console.info('outputFilePath:', outputFilePath)
+
+  const writer = csvWriter.createObjectCsvWriter({
+    path: outputFilePath,
+    header: [
+      { id: 'passport_id', title: 'passport_id' },
+      { id: 'owner_address', title: 'owner_address' },
+      { id: 'discourse_username', title: 'discourse_username' }
+    ]
+  })
+  const csvRows: any[] = []
```

Comment:
At some point we should create a type for the Row, we have one anyway!

Reply:
@johnmark13 Yes, created an issue for it here:  https://github.com/nation3/nationcred-datasets/issues/90